### ISSUE TITLE
Platform dependence

### ISSUE THREAD

**Issue by benfulcher** [chlubba/catch22#10]
Testing has revealed differences in _catch22_ results across platforms/compilers (mac/windows/linux)/(Clang/gcc)

**Comment by imraniac** [comment]
The problem might be in `CO_AutoCorr.c`, there are two definitions of the complex data type which depends on the compiler.
```
#if defined(__GNUC__) || defined(__GNUG__)
typedef double complex cplx;
#elif defined(_MSC_VER)
typedef _Dcomplex cplx;
```
There could be an implementation bug which needs to be resolved

**Comment by chlubba** [comment]
Hi @imraniac,

thank you for taking a look into the code. You are right that this compiler-depencency is the source of differences between platforms. 

Unfortunately, it is to our knowledge unavoidable to branch on the type of compiler because the implementation of complex numbers differs between GCC (Linux) and the MS Visual Studio compiler (Windows). The complex datatype and the associated methods are different between these compilers. Therefore, no single, identical version of the code works for both. I.e. if you write the code to work for GCC, it won't run on Windows and vice versa. The only solution so far to make catch22 compile using either compiler, are these preprocessor-branches you identified and two different versions of the C-implementation, one for each compiler.

And apparently, some complex number computations differ between the two catch22 versions. The differences most likely affect the FFT, accordingly the `co_autocorr` function, and possibly other functions.

Suggestions on how to handle the complex data-type in a unified way across platforms are very welcome! Otherwise the two different branches of the C-implementation have to become better aligned to behave exactly identically.

Carl

**Comment by chlubba** [comment]
This issue should now be fixed in release v0.2.0 onwards.
